packages/block-editor/src/components/inspector-controls/index.native.js

```diff
@@ -30,7 +39,9 @@ const FillWithSettingsButton = ( { children, ...props } ) => {
 					</BottomSheetConsumer>
 				}
 			</Fill>
-			{ React.Children.count( children ) > 0 && <BlockSettingsButton /> }
+			{ React.Children.toArray( children ).filter(
+				( child ) => child.props.excludeFromSettings !== true
```

Comment:
I was wondering if we could try to avoid this change since it's just for one use case. Maybe we could just handle this in Cover itself (I left some comments below explaining how we could do this)

Reply:
Thank you for the review and the precious feedback @geriux 🙇‍♂️ 
I tried to avoid touching the InspectorControls following your advice and 2-3 other ways (e.g. [this one](https://github.com/WordPress/gutenberg/pull/23870/commits/537211c677c8fa11667012e97615cbd007b8c61f)) but I always end up with some minor status inconsistency/bug. As a solution I ended up with a cleaner approach in the InspectorControls. Wdyt?

Reply:
Happy to review and give feedback =)

Thanks for making those changes, although I still think we could avoid adding an extra prop to the `InspectorControls` component.

With this approach, there's still an issue ([related to this](https://github.com/WordPress/gutenberg/pull/23870#discussion_r459291688)).

The `isCustomColorPickerShowing` value is not being updated when the modal is being closed when tapping outside. Same thing if I use the back button on an Android device. See the gif below:

<kbd><img src="https://user-images.githubusercontent.com/4885740/88288267-cc9a3980-ccf3-11ea-8c62-8ff7c11b6329.gif" width="250"/></kbd>

Reply:
Thanks again for all your help and feedback @geriux :)
I followed your advice and backed off from touching the inspector controls.